iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -66,8 +81,36 @@ private static async Task RunSampleAsync()
             // -> Periodically send "temperature" over telemetry.
             // -> Send "maxTempSinceLastReboot" over property update, when a new max temperature is set.
 
+            if (string.IsNullOrWhiteSpace(s_deviceSecurityType))
+            {
+                throw new ArgumentNullException("Device security type needs to be specified, please set the environment variable \"IOTHUB_DEVICE_SECURITY_TYPE\".");
+            }
+
             s_logger.LogDebug($"Initialize the device client.");
-            InitializeDeviceClientAsync();
+            switch (s_deviceSecurityType.ToLowerInvariant())
+            {
+                case "dps":
+                    if (ValidateArgsForDpsFlow())
+                    {
+                        DeviceRegistrationResult dpsRegistrationResult = await ProvisionDeviceAsync();
+                        var authMethod = new DeviceAuthenticationWithRegistrySymmetricKey(dpsRegistrationResult.DeviceId, s_deviceSymmetricKey);
+                        InitializeDeviceClientAsync(dpsRegistrationResult.AssignedHub, authMethod);
+                        break;
+                    }
+                    throw new ArgumentException("Required environment variables are not set for DPS flow, please recheck your environment.");
```

Comment:
I am thinking about changing this to log error and then exit (as opposed to throwing an exception). Thoughts?